### PR TITLE
Extracted code repetitive code into methods (using enums to their potential)

### PR DESCRIPTION
**Pull request by viveanban**

### CODE REVIEW

File: src/main/java/ca/onepoint/yul/classes/ManifestationManagement.java [hunk 1]
```diff
@@ -12,54 +12,97 @@
...
+        MapDto map = getFirstMap(mapService);
+
+        if (map == null) {
+            return new ArrayList<>();
+        }
+
+        square = map.getSquare();
+
+        protestorsBlockVerticalStreet = new Random().nextBoolean(); // to determine protestors direction
+        randomStreetIndex = new Random().nextInt(MAP_SIZE); // random street
```
Comment (ferdousara-parvin) [inline src/main/java/ca/onepoint/yul/classes/ManifestationManagement.java]: unnecessary comment

File: src/main/java/ca/onepoint/yul/dto/SquareDto.java [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+import ca.onepoint.yul.classes.SquareType;
 import lombok.Data;
 
 @Data
 public class SquareDto {
     private Integer value;
     private String image;
+
+    public boolean isRoute(){
+        return   value == SquareType.ROUTE.value;
```
Comment (ferdousara-parvin) [inline src/main/java/ca/onepoint/yul/dto/SquareDto.java]: unwanted space béw return and value

File: src/main/java/ca/onepoint/yul/classes/ManifestationManagement.java [hunk 3]
```diff
@@ -12,54 +12,97 @@
...
-                if (square[isColumn ? varyingIndex : rowOrColumnIndex][isColumn ? rowOrColumnIndex : varyingIndex].getValue() == 1) {
-                    int blockLength = getLengthConsecutiveBlocks(varyingIndex, rowOrColumnIndex, isColumn, square);
-
-                    // If there is more than 1 square of route consecutively
-                    if (blockLength > 1) {
-                        ArrayList<AvatarDto> addedAvatars = createProtestors(varyingIndex, rowOrColumnIndex, blockLength, isColumn);
-                        allProtesters.addAll(addedAvatars);
-                        return allProtesters;
-                    }
+    private static boolean protestorsCanExist(int startIndex) {
```
Comment (ferdousara-parvin) [inline src/main/java/ca/onepoint/yul/classes/ManifestationManagement.java:84]: rename to _manifestationCanHappen_? something along those lines..

File: src/main/java/ca/onepoint/yul/classes/ManifestationManagement.java [hunk 4]
```diff
@@ -12,54 +12,97 @@
...
+        for (int varyingIndex = startIndex; varyingIndex < MAP_SIZE; varyingIndex++) {
+            int nextY = protestorsBlockVerticalStreet ? varyingIndex : randomStreetIndex;
+            int nextX = protestorsBlockVerticalStreet ? randomStreetIndex : varyingIndex;
+            if (square[nextY][nextX].isRoute()) {
+                int blockLength = getLengthConsecutiveBlocks(varyingIndex, protestorsBlockVerticalStreet);
+
+                // If there is more than 1 square of route consecutively
```
Comment (ferdousara-parvin) [inline src/main/java/ca/onepoint/yul/classes/ManifestationManagement.java]: complete this comment by saying something like "then manifestation can occur"

**Review by ferdousara-parvin — APPROVED** [review]

**Review by Hambrsoom — APPROVED** [review]

**Pull request merged**